src/test/ui/consts/const-eval/const_let.rs

```diff
@@ -9,10 +9,13 @@ impl Drop for FakeNeedsDrop {
 // ok
 const X: FakeNeedsDrop = { let x = FakeNeedsDrop; x };
 
+// ok (used to incorrectly error, see #62273)
+const X2: FakeNeedsDrop = { let x; x = FakeNeedsDrop; x };
```

Comment:
Can you add a version with `mut` to ensure there's no funny business in the future?

Reply:
What do you have in mind? `Y` but with separate initialization?

Reply:
maybe `let mut x = None; x = Some(FakeNeedsDrop); x` ?

Reply:
@oli-obk's test case is better than mine; I just wanted `{ let mut x; x = FakeNeedsDrop; x };` =P

Reply:
Your version would just cause an "unused `mut`" warning, you need multiple assignments for anything else to be different.

Reply:
I didn't expect it to be different; the point was to exercise more of the surface, not the semantics.